src/Common/src/System/Net/Internals/SocketException.Windows.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Net.Sockets;
+
+namespace System.Net.Internals
+{
+    internal class InternalSocketException : SocketException
+    {
+        // TODO: Add a public ctor to SocketException instead.
```

Comment:
We want HResult to be correctly set in Unix implementations. More tests are needed but AFAIK that's correctly set by Win32Exception for Windows cases.
/cc: @pgavlin